### PR TITLE
Done_HW_12.2_Beheka Misha

### PR DESCRIPTION
**Pull request by MishaBeheka**

### CODE REVIEW

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 1]
```diff
@@ -19,7 +25,19 @@
...
+            case FULL:
+                return String.valueOf(LocalDate.now());
+            case YEAR:
+                return String.valueOf(LocalDate.now().getYear());
+            case MONTH:
+                return String.valueOf(LocalDate.now().getMonth());
+            case DAY:
+                return String.valueOf(LocalDate.now().getDayOfMonth());
+            default:
+                return "";
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: `В любом другом случае бросить DateTimeException`

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 2]
```diff
@@ -31,39 +49,43 @@ public String todayDate(DateTimePart datePart) {
      *                   - 3-й элемент массива - день (число);
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        return (dateParams.length != 0 && dateParams[0] > 1970
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: simplify this method: return Optional.of(...) 
and let's use try-catch block here

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 3]
```diff
@@ -74,16 +96,23 @@ public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
      * - "someDate is today" - если someDate - сегодня
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        return someDate.isAfter(LocalDate.now()) ? someDate + " is after " + LocalDate.now()
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: create a variable in this method not to call LocalDate.now multiple times

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 4]
```diff
@@ -96,23 +125,36 @@ public String beforeOrAfter(LocalDate someDate) {
      * OffsetDateTime советуют использовать при записи даты в базу данных.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return OffsetDateTime.of(localTime, ZoneOffset.of("+02:00"));
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: let's create a constant from "+02:00"

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 5]
```diff
@@ -96,23 +125,36 @@ public String beforeOrAfter(LocalDate someDate) {
...
     }
 
     /**
      * Дана строка в виде "yyyymmdd".
      * Необходимо вернуть Optional даты в LocalDate формате
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        try {
+            int year = Integer.parseInt(date.substring(0, 4));
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: let's use standard DateTimeFormatter

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 6]
```diff
@@ -96,23 +125,36 @@ public String beforeOrAfter(LocalDate someDate) {
...
     }
 
     /**
      * Дана строка в виде "d MMM yyyy" (MMM - Sep, Oct, etc).
      * Необходимо вернуть Optional даты в LocalDate формате
      */
     public Optional<LocalDate> customParseDate(String date) {
-        return Optional.empty();
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMM yyyy", Locale.UK);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: create a constant from "d MMM yyyy"

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 7]
```diff
@@ -123,6 +165,6 @@ public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
      * или сообщение "dateTime can't be formatted!"
      */
     public String formatDate(LocalDateTime dateTime) {
-        return "Date can't be formatted!";
+        return dateTime.format(DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm", Locale.UK));
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java]: the same

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 1 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 2 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 3 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 4 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 5 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 6 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 7 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java]: Done

**Pull request closed without merging**